src/Indicator.vala

```diff
@@ -18,15 +18,27 @@
 public class BluetoothIndicator.Indicator : Wingpanel.Indicator {
     public bool is_in_session { get; construct; default = false; }
 
-    private Widgets.PopoverWidget popover_widget;
-    private Widgets.DisplayWidget dynamic_icon;
+    public signal void device_requested (BluetoothIndicator.Services.Device device);
+    private BluetoothIndicator.Services.Device paired_device;
+    BluetoothIndicator.Widgets.PopoverWidget popover_widget;
+    Widgets.DisplayWidget? display_widget;
     private Services.ObjectManager object_manager;
 
     public Indicator (bool is_in_session) {
         Object (
             code_name: Wingpanel.Indicator.BLUETOOTH,
             is_in_session: is_in_session
         );
+
+        display_widget = new Widgets.DisplayWidget (object_manager);
```

Comment:
You'll need to call update_tooltip () here to initialise your tooltip. Otherwise, the user will only see a tooltip when they pair a device.

```suggestion
        var settings = new Settings ("io.elementary.desktop.wingpanel.bluetooth");
        display_widget = new Widgets.DisplayWidget (object_manager);
        
        var state = settings.get_boolean ("bluetooth-enabled");
        update_tooltip (state, false);
```